lib/scanner/walk.go

```diff
@@ -515,8 +510,13 @@ func (w *walker) normalizePath(path string, info fs.FileInfo) (normPath string,
 	return normPath, false
 }
 
-// updateFileInfo updates walker specific members of protocol.FileInfo that do not depend on type
-func (w *walker) updateFileInfo(file, curFile protocol.FileInfo) protocol.FileInfo {
+// updateFileInfo updates walker specific members of protocol.FileInfo
+func (w *walker) updateFileInfo(file, curFile protocol.FileInfo, hasCurFile bool) protocol.FileInfo {
```

Comment:
No need to pass the `hasCurFile` really. If we didn't have it, the permissions on the passed `curFile` would be zero and the `|` would be a no-op. But it doesn't matter greatly either way.